guides/customizing-component-styles.md

```diff
@@ -36,6 +37,20 @@ important to keep in mind, since even using high specificity and shadow-piercing
 not target elements that are not direct children of your component. Global styles are recommended
 for targeting such components.
 
+##### Customizing Themed Components
+
+Component styles which are dynamic and created by your Angular Material Theme behave differently than 
+other styles.  These styles are compiled from `styles.scss` (and any files imported into it), and 
+inserted directly into the DOM via `<style>` tags in the `<head>` of your document. This creates an
+additional challenge for modifying these styles, as they will override your individual component styles
+unless you use more specific selectors than Angular Material.   In practice, the most effective way to modify
```

Comment:
> as they will override your individual component styles
> unless you use more specific selectors than Angular Material

I don't quite follow this. Because Angular Material doesn't use view encapsulation, the theme styles and the component styles should be on equal footing with respect to overrides. And the point about the theme styles applying unless you use a more specific selector- that's really just a basic of how CSS works, which I don't think warrants an additional mention here.

Reply:
My goal here is to provide clarity about how the themed styles produced by the Material theme mixin differ from normal material styles.   Because they are inserted into the head, you cannot override them with an equally-specific selector and being lower in the css cascade.   For example, you can't override a Material theme style using !important in a component style file.  Surpassing material's specificity is also very challenging because you do not have access to the individual elements to add ids to them.     Those approaches -do- work for non-themed Material styles (because they end up in the styles.scss in the live environment, and not in the DOM)

Because the themed css works differently than the rest of Material's css, I wanted to highlight that difference.

It's possible there's a better and clearer way of explaining this particular concern.

Reply:
> . For example, you can't override a Material theme style using !important in a component style file.

That's not quite right: https://stackblitz.com/edit/angular-wrts87?file=src/app/button-overview-example.css

The only thing that would prevent a component-level style from applying is specificity. For ordering, when two definitions have the specificity, the latter one wins. What you might have seen in the past is that a component-level style wouldn't apply to the internals of an Angular Material component due to view encapsulation. This would make it so that styles in the component only apply to elements that directly live within that component. 

As for overcoming the component's specificity- yes, this is not easy. We do try to always [use the lowest specificity possible](https://github.com/angular/components/blob/master/CODING_STANDARDS.md#use-lowest-specificity-possible), but sometimes we're limited in what's actually possible. While we accept that some degree of overriding is inevitable, we do try to discourage it as much as we can, since it makes updates to the components more challenging. (This whole thing is more of a problem for us inside Google, though, since we're responsible for fixing any downstream applications when we make CSS changes).